src/GeniusYield/Types/Providers.hs

```diff
@@ -185,6 +191,35 @@ type GYLookupDatum = GYDatumHash -> IO (Maybe GYDatum)
 -- | How to submit a transaction?
 type GYSubmitTx = GYTx -> IO GYTxId
 
+-------------------------------------------------------------------------------
+-- Await tx confirmation
+-------------------------------------------------------------------------------
+
+-- | How to await for a transaction confirmation?
+type GYAwaitTx = GYAwaitTxParameters -> GYTxId -> IO ()
+
+-- | Await transaction parameters
+data GYAwaitTxParameters = GYAwaitTxParameters
+                           { maxAttempts   :: Int
+                           -- ^ Max number of attempts before give up.
+                           , checkInterval :: Int
+                           -- ^ Wait time for each attempt (in nanoseconds).
+                           , confirmations :: Int
+                           -- ^ Min number of block confirmation.
+                           }
+    deriving stock (Show)
+
+instance Default GYAwaitTxParameters where
+    def = GYAwaitTxParameters
+          { maxAttempts   = 10
+          , checkInterval = 3_000_000
+          , confirmations = 1
+          }
+
+newtype GYAwaitTxException = GYAwaitTxException GYAwaitTxParameters
+  deriving stock (Show)
```

Comment:
I think it's show instance could be bit more informative, like it could say that "Tried exceeded given maximum" and then the say the parameters. Though not a big deal.